session/auth/authprovider/authprovider.go

```diff
@@ -173,6 +181,23 @@ func (ap *authProvider) VerifyTokenAuthority(ctx context.Context, req *auth.Veri
 	return &auth.VerifyTokenAuthorityResponse{Signed: sign.Sign(nil, req.Payload, priv)}, nil
 }
 
+func (ap *authProvider) getAuthConfig(host string) (*types.AuthConfig, error) {
+	if _, exists := ap.authConfigCache[host]; !exists {
```

Comment:
This map access is not safe as read and write could be happening at the same time.

Reply:
Ah, I mistakenly thought that so long as only one thread was writing at a time it would be safe. I will move the mutex.